### PR TITLE
Fully functional docker with reverse-proxy

### PR DESCRIPTION
**Pull request by pawelrubin**
- Faster docker-compose due to using pip instead of pipenv through the process;
- Using nginx reverse-proxy to connect externally to frontend and backend on the same port;
- Small fixes due to changes in backend/frontend structure.

~ @evergreenexplicit

### CODE REVIEW

File: docker/Dockerfile.backend [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 
-COPY backend/Pipfile Pipfile
-COPY backend/Pipfile.lock Pipfile.lock
+RUN pipenv lock --requirements > requirements.txt
 
-RUN pipenv install --dev
+RUN cat requirements.txt 
 
-EXPOSE 80
+RUN pip install -r requirements.txt
```
Comment (wadamek65) [inline docker/Dockerfile.backend:14]: If you don't like `pipenv` another popular alternative to `pipenv` is `poetry`. It seems to be much faster and many say they like it more. Haven't used it myself but if you feel like it you can play around with it.

File: docker/Dockerfile.frontend [hunk 2]
```diff
@@ -3,6 +3,4 @@ WORKDIR /app
 ENV PATH /app/node_modules/.bin:$PATH
 COPY frontend/package.json /app/package.json
 COPY frontend/package-lock.json /app/package-lock.json
-RUN npm install --silent
-RUN npm install react-scripts@3.0.1 -g --silent
-RUN npm install node-sass@latest
+RUN npm install
```
Comment (wadamek65) [inline docker/Dockerfile.frontend:6]: You should usually run this with `--production` flag so no dev dependencies are installed. Also keep in mind that `npm install` can overwrite `package-lock` if there are newer packages versions than specified in `package.json`. It is better to use a dependency manager that respects lockfiles like `yarn`. This problem probably won't affect you but it's good to know.

File: docker/docker-compose.yml [hunk 3]
```diff
@@ -32,10 +44,31 @@ services:
...
+      - "82:82"
     environment:
+      - HOST=${HOST}
+      - PORT=82
       - NODE_ENV=development
       - CHOKIDAR_USEPOLLING=true
+
+
+
+  proxy:
```
Comment (wadamek65) [inline docker/docker-compose.yml:60]: very inconsistent formatting in this file. Too many unnecessary new lines throughout the whole file.

**Review by wadamek65 — APPROVED** [review]

File: docker/Dockerfile.frontend (hunk 2 above)
Comment (konrad-grochowski) [reply docker/Dockerfile.frontend:6]: This build is development-oriented, though I will take this into account while making the production one. Thanks!

**Review by konrad-grochowski — APPROVED** [review]

**Pull request merged**